K2Bridge/Models/Response/Aggregations/BucketAggregate.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Models.Response.Aggregations
+{
+    using System.Collections.Generic;
+    using K2Bridge.JsonConverters;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Describes bucket aggregate response element.
+    /// </summary>
+    [JsonConverter(typeof(BucketAggregateConverter))]
+    public class BucketAggregate<TBucket> : IAggregate
```

Comment:
Just a thought - do we need this to be generic?
It complicates the code without any real benefit (unless the serialization needs it?).
Like if instead of being generic, it had a `List<IKeyedbucket>`, and IKeyedBucket had an `object Key` property, you could use this class more easily,
for example in `BucketsAggregateConverter`, you would not need the generic method and could replace the if-else with one call.

But maybe there is a reason for this?

Reply:
Yes. I agree. Initially, I wanted to avoid using 'object' to ensure we check the type. And, the second point, it seems some elastic search buckets are not always inheriting KeyedBucket. Anyway, as I am not sure we will implement them, we can remove this complexity at the moment. Fixed.